### DISCUSSION TITLE
Carousel autoplay - get working within a collapsable css/ js element

### DISCUSSION THREAD

**Discussion by v001-ali** [fancyapps/ui#439] [Q&A] [answered]
Hey,

I would like to ask for some help, Im trying to place a fancy apps carousel within a collapsable.

I have edited a CodePen link which I used to illustrate my problem, essentially the carousel itself works but placing it within a collapsable menu seems to break the autoplay feature, I imagine this is something I have done incorrectly or overlooked something, but I thought id ask for help as I'm struggling with how to fix.  Im linking to both examples below.

Collapsable Code link 
https://codepen.io/aliv001/pen/ExeRZbx
Original Code link
https://codepen.io/fancyapps/pen/qBMorXM

**Comment by fancyapps** [comment, marked as answer]
Hi,

If you render Carousel in a hidden element, you must call the 'updateMetrics()' method after you make it visible. Also you need to disable automatic start of "Autoplay", just start/stop when you change Carousel visibility

Demo - https://codepen.io/fancyapps/pen/ZEMRRya?editors=0010

**Reply by v001-ali** [reply]
Thanks this works perfectly, thanks for the code as well!
